libs/FirmwareKit/source/FirmwareKit.cpp

```diff
@@ -46,10 +46,18 @@ auto FirmwareKit::loadUpdate(const Version &version) -> bool
 {
 	auto path = getPathOfVersion(version);
 
-	return loadUpdate(path);
+	return load(path);
 }
 
-auto FirmwareKit::loadUpdate(const std::filesystem::path &path) -> bool
+auto FirmwareKit::loadFactory() -> bool
+{
+	if (auto factory_loaded = load(_config.factory_path); factory_loaded) {
+		return true;
+	}
+	return loadUpdate(Version {.major = 1, .minor = 0, .revision = 0});
```

Comment:
I'm not sure I understand what is happening here.

if you can load the factory image, you return true meaning that it's the one who is going to be installed, right?

but if you can't, why would you load an update instead?